### ISSUE TITLE
kbld pkg should retry on transient network failures

### ISSUE THREAD

**Issue by cppforlife** [k14s/kbld#8]
change should possibly be in the underlying go-containerregistry library.

```
--- FAIL: TestPkgUnpkgSuccessful (39.76s)
    kbld.go:72: Failed to successfully execute 'kbld package -f - --output /tmp/kbld-test-pkg-unpkg-successful --yes': Execution error: stdout: '' stderr: 'package | exporting 2 images...
        package | will export index.docker.io/library/redis@sha256:000339fb57e0ddf2d48d72f3341e47a8ca3b1beae9bdcb25a96323095b72a79b
        package | will export gcr.io/cloud-builders/gcs-fetcher@sha256:055519529bf1ba12bf916fa42d6d3f68bdc581413621c269425bb0fee2467a93
        package | exported 2 images
        Error: Getting compressed layer: Get https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/97/97a473b37fb2921176dcdeb10cecd0171a8b2ef20ea51fcbf330a8ccd9c7efb3/data?verify=1564597134-8w3hML1hGiv1%2Fwzqpms%2BhwqDGXo%3D: net/http: timeout awaiting response headers
        ' error: 'exit status 1'
```

**Comment by ciriarte** [comment]
@cppforlife I want to second this issue. It is painful in (potentially) long operations such as package/unpackage.

**Comment by cppforlife** [comment]
fixed in https://github.com/k14s/kbld/commit/509761e690a53ba7772aae238ae2d1606cde38fe

**Comment by ciriarte** [comment]
@cppforlife is there any reason why this issue is still open?

**Comment by cppforlife** [comment]
fix available in v0.24.0
